### ISSUE TITLE
InnerFire (BB #521)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#521]
This issue was migrated from bitbucket.
**Original Reporter:** doomscaler
**Original Date:** 02.03.2014 19:11:56 GMT+0000
**Original Priority:** critical
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521

<hr>

is there a solution to a bug with InnerFire the charges fall under the stress on the shield

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 02.03.2014 19:39:08 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8870460

<hr>

Can you try the patch below and report back if it works? :)

```
#!c++

# HG changeset patch
# User LordUsagi
# Date 1379123140 10800
#      Fri Sep 13 22:45:40 2013 -0300
# Node ID 6b24789318be519499af3dd357b5f430c0e90623
# Parent  e8fa64aa13d866c61aa1a3a403d960773a5fa1f9
[Spells/Items] proc_ex_absorb fix

diff -r e8fa64aa13d8 -r 6b24789318be src/game/Player.cpp
--- a/src/game/Player.cpp   Fri Sep 13 22:38:51 2013 -0300
+++ b/src/game/Player.cpp   Fri Sep 13 22:45:40 2013 -0300
@@ -7336,7 +7336,7 @@
 void Player::CastItemCombatSpell(Unit *target, WeaponAttackType attType, uint32 procVictim, uint32 procEx, Item *item, ItemPrototype const * proto, SpellEntry const *spell)
 {
     // Can do effect if any damage done to target
-    if (procVictim & PROC_FLAG_TAKEN_ANY_DAMAGE)
+    if (procVictim & PROC_FLAG_TAKEN_ANY_DAMAGE || procVictim & PROC_EX_ABSORB)
     {
         for (uint8 i = 0; i < 5; ++i)
         {

```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 02.03.2014 19:41:17 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8870479

<hr>

I doubt it'll work now that I'm looking back at it, but I have other possible solutions.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** doomscaler
**Original Date:** 02.03.2014 19:46:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8870520

<hr>

but you can see them?)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 02.03.2014 19:47:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8870527

<hr>

See what?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.03.2014 02:11:03 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8900260

<hr>

```
#!c++

# HG changeset patch
# User Nick@Nick-PC.BigPond
# Date 1341331871 -36000
#      Wed Jul 04 02:11:11 2012 +1000
# Node ID dd32d2fda118d410aa0a55259463d68c4d9d2b5f
# Parent  5df81a907f1157af14d1395e282bca93bf1d9026
Fix for Inner Fire proccing when shielded

diff -r 5df81a907f11 -r dd32d2fda118 src/game/Unit.cpp
--- a/src/game/Unit.cpp Tue Jul 03 23:58:21 2012 +1000
+++ b/src/game/Unit.cpp Wed Jul 04 02:11:11 2012 +1000
@@ -10683,7 +10683,7 @@
     {
         SpellProcEventEntry const* spellProcEvent = NULL;
         bool active = (damage > 0) || (procExtra & PROC_EX_ABSORB && isVictim);
-        if (!IsTriggeredAtSpellProcEvent(pTarget, itr->second, procSpell, procFlag, procExtra, attType, isVictim, active, spellProcEvent))
+        if (!IsTriggeredAtSpellProcEvent(pTarget, itr->second, procSpell, procFlag, procExtra, attType, isVictim, (bool)damage, spellProcEvent))
            continue;

         procTriggered.push_back(ProcTriggeredData(spellProcEvent, itr->second));

```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.03.2014 02:11:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8900269

<hr>

The above code may cause bugs BUT I need you to find them out fr me.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.03.2014 02:46:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/521/#comment-8900539

<hr>

[Core/Spells] Fix issues with proccing on absorbs. Closes issue #521

→ <<cset 97314f11cc57>>
